core/common/src/math.kt

```diff
@@ -200,4 +200,17 @@ internal fun multiplyAndAdd(d: Long, n: Long, r: Long): Long {
         mr -= n
     }
     return safeAdd(safeMultiply(md, n), mr)
-}
\ No newline at end of file
+}
+
+// org.threeten.bp.chrono.IsoChronology#isLeapYear
+internal fun isLeapYear(year: Int): Boolean {
```

Comment:
This file contains common math functions, not specific to any date calculations. I suggest to find another file for these new ones, or introduce a new one, e.g. dateMath.kt/calendarMath.kt/IsoChronology.kt

Reply:
It does define some constants like `internal const val HOURS_PER_DAY = 24` or `NANOS_PER_DAY` already, so it's not just pure math here. I don't think it's worth it yet to split the file, because
* it's relatively short,
* when writing code, I like to look over the available constants, and seeing all of them in one place helps, whereas we would need to move, say, `HOURS_PER_DAY` into the date-specific-math file, keeping `NANOS_PER_ONE` in the pure math one.

Reply:
Now, when we've got two new functions, it's a good chance to group these constants together with them in a separate file.
NANOS_PER_ONE can also be seen as a time-related constant, not general math.

Reply:
Peculiarly, trying to implement this refactoring slows down the code on JS considerably. The test for `fromEpochDays` now fails, and checking on my machine, before the refactoring, it takes 1.2 ± 0.2 seconds, while after the refactoring, it takes 1.6 ± 0.3 seconds. I propose that we revert this change.

Reply:
Can't confirm such effect, on the opposite the test runs faster for me on the latest commit than on the previous.
I compared generated JS code and didn't find anything suspicious in the difference.

Reply:
As the last resort, the timeout could be increased if it's essential to test all these values.

Reply:
Yeah, let's do that.